backend/molgenis-emx2-sql/src/main/java/org/molgenis/emx2/sql/SqlTable.java

```diff
@@ -173,15 +173,16 @@ private static void truncateTransaction(
       String mg_table = t.getMgTableClass(t.getMetadata());
       // cascading delete will take care of subclass deletes
       database
-          .getJooq()
+          .getJooqWithExtendedTimeout()
           .deleteFrom(rootTable.getJooqTable())
           .where(field(MG_TABLECLASS).equal(mg_table))
           .execute();
     }
     // else in normal table simply call delete
     else {
       // truncate would be faster, but then we need add code to remove and re-add foreign keys
-      database.getJooq().deleteFrom(t.getJooqTable()).execute();
```

Comment:
ah, this is indeed not a truncate but a delete. So instead of prolonging we could do this in batches.
However, that would also require a select. 

docs suggest that we could do dsl.truncate("your_table").cascade().execute(); instead. But I am not sure.

Bottomline: I think deleteFrom is a better option for now.